misc/package/Data Files/MWSE/core/mcm/components/Component.lua

```diff
@@ -239,4 +239,9 @@ function Component:create(parentBlock)
 	end
 end
 
+--- This dummy method was added so that `Category` and `Template` can call `addDefaultsToDescriptions`
+-- indiscriminantly on child elements. Also it makes the documentation a little cleaner.
+function Component:addDefaultsToDescriptions(defaultConfig)
+end
+
```

Comment:
Is it needed to declare the method on the parent `Component` class? To me, it looks like this can be declared the Category component.

Reply:
I'm not really sure that this is the best solution. I just found it made the implementation a bit easier since it means I didn't have to add a check to make sure `component.componetType` is a category or setting or page before calling the method. But I think that the ability to call it on a component probably should not be included in the documentation, and just kept as a secret implementation detail.

But maybe it's better to just explicitly check before calling it each time?